src/utils/utils.ts

```diff
@@ -145,6 +147,39 @@ export class Utils {
     return true;
   }
 
+  public static async getTarget(): Promise<ITarget> {
+    return await cfGetTarget();
+  }
+
+  public static async isLoggedInToCf(): Promise<boolean> {
+    try {
+      const target: ITarget = await this.getTarget();
+      // user is connected to CF
+      if (
+        !isEmpty(target.user) &&
+        !isEmpty(target.org) &&
+        !isEmpty(target.space) &&
+        !isEmpty(target["api endpoint"])
+      ) {
+        return true;
+      }
+    } catch (e) {
+      // User is not logged in to Cloud Foundry
+    }
+    return false;
+  }
+
+  public static async isLoggedInToCfWithProgress(): Promise<boolean> {
+    return window.withProgress(
+      {
+        location: ProgressLocation.Notification,
+        title: "Checking your connection to Cloud Foundry...",
```

Comment:
KM (Paola)? candidate to extract in static messages ?

Reply:
asked Paola/Shira to review

Reply:
Paola approved